keps/sig-windows/4802-windows-node-shutdown/README.md

```diff
@@ -0,0 +1,545 @@
+
+We will also add function of "UpdatePreShutdownInfo" and "QueryPreShutdownInfo" to Windows service package, these functions will be 
+used by the "nodeshutdown_manager" to query and update the preshtudown timeout.
+
+If we will take the 'pre-shutdown timeout value as pariale with the 'InhibitDelayMaxSec` in linux,
+then the other parts of implementation will be pretty much similar with what linux has [implemented](https://github.com/kubernetes/enhancements/tree/master/keps/sig-node/2000-graceful-node-shutdown#implementation)
+
+The poc of the implementation can be found [here](https://github.com/zylxjtu/kubernetes/commit/854ea4bde88c0905241b43f5f80d470967bb909f#diff-8494875e9a1e884afd36f32fa90dceb2b827616f1e981ecb83a12901262214c7)
+
+### Notes/Constraints/Caveats (Optional)
+
+Third patry service manager on Windows (other than the native sc.exe) such as nssm.exe support log redirection and log file rotation, 
+which will be missed if running kublet itself as a native service and loading through sc.exe. 
+
+The logging of kublet is out of the scope of this KEP. But [kube-log-runner](https://pkg.go.dev/k8s.io/component-base/logs/kube-log-runner) has been trying to address the k8s log redirection  
+and there has been efforts to enhance the kube-log-runner with optional log file rotation and clean up at [here](https://github.com/kubernetes/kubernetes/pull/127667)
+
+### Risks and Mitigations
+
+Please refer to the the [part](https://github.com/kubernetes/enhancements/tree/master/keps/sig-node/2000-graceful-node-shutdown#notesconstraintscaveats-optional) in KEP-2000
+
+## Design Details
+
+Use the same Kubelet Config as in linux.
+```
+type KubeletConfiguration struct {
+    ...
+    ShutdownGracePeriod metav1.Duration
+    ShutdownGracePeriodCriticalPods metav1.Duration
+}
+```
+
+Communication with service control manager will make use of golang.org/x/sys/windows/svc packag,
+which is already included in [vendor](https://github.com/kubernetes/kubernetes/tree/release-1.19/vendor/golang.org/x/sys/windows)
+
+
+Termination of pods will make use of the existing
+[killPod](https://github.com/kubernetes/kubernetes/blob/release-1.19/pkg/kubelet/pod_workers.go#L292) function
+from the `kubelet` package and specify the appropriate `gracePeriodOverride` as
+necessary.
+
+### Test Plan
+
+[x] I/we understand the owners of the involved components may require updates to
+existing tests to make this code solid enough prior to committing the changes necessary
+to implement this enhancement.
+
+##### Prerequisite testing updates
+
+
+##### Unit tests
+
+It does not seem possible to unit test the Windows logic of setting service handler and update registry keys
+
+The main logic of kill pod is the same as what linux did and there has been unit tests covered there.
+
+##### Integration tests
+
+Node level integration test will be added later when the node e2e test framework is available for Windows node
+
+- <test>: <link to test coverage>
```

Comment:
Can you add something saying we don't currently have an automated way to run integration tests for Windows specific code and we will cover all the scenarios with e2e tests?